riscv-runtime/src/commit.rs

```diff
@@ -0,0 +1,55 @@
+pub static mut PUBLICS: CommittedPublics = CommittedPublics::new();
+
+pub fn commit(n: u32) {
+    unsafe { PUBLICS.commit(n) }
+}
+
+pub(crate) fn finalize() -> [u64; 4] {
+    unsafe { PUBLICS.finalize() }
+}
+
+pub struct CommittedPublics {
+    state: [u64; 12],
```

Comment:
Why 12?

Reply:
That's the size of the poseidon state which we replicate here in this pseudo-sponge construction